tket/src/Predicates/Predicates.cpp

```diff
@@ -525,6 +526,45 @@ std::string MaxNQubitsPredicate::to_string() const {
   return auto_name(*this) + "(" + std::to_string(n_qubits_) + ")";
 }
 
+bool MaxNClRegPredicate::verify(const Circuit& circ) const {
+  bit_vector_t all_bits = circ.all_bits();
+  std::set<std::string> bits_set;
+  for (Bit bit : all_bits) {
+    auto it = bits_set.find(bit.reg_name());
+    if (it == bits_set.end()) {
+      bits_set.insert(bit.reg_name());
+    }
```

Comment:
```suggestion
    bits_set.insert(bit.reg_name());
```
No need to check if it's there already, this is automatic for sets.

Reply:
Thank you, good point!